.ci/cico_build_and_deploy_custom_rh-che.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+set -x
+set -e
+set +o nounset
+
+/usr/sbin/setenforce 0
+
+function retrieveActiveToken() {
+  CURL_OUTPUT=$(curl -H "Content-Type: application/json" -X POST -d '{"refresh_token":"'$KEYCLOAK_TOKEN'"}' https://auth.prod-preview.openshift.io:8443/api/token/refresh)
+  ACTIVE_TOKEN=$(echo $CURL_OUTPUT | jq --raw-output ".token | .access_token")
+  if [[ -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_USERNAME}" ]]; then
+    empty_credentials="OSIO username is empty, "
+  fi
+  if [[ -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_PASSWORD}" ]]; then
+    empty_credentials=${empty_credentials}"OSIO password is empty, "
+  fi
+  if [[ -z "${ACTIVE_TOKEN}" ]]; then
+    empty_credentials=${empty_credentials}"Keycloak token is empty"
+  fi
+  if [[ ! -z "${empty_credentials}" ]]; then
+    echo ${empty_credentials}
+    exit 1
+  fi
+  if [[ $(curl -X GET -H "Authorization: Bearer ${ACTIVE_TOKEN}" https://auth.prod-preview.openshift.io:8443/api/token?for= \
+     |  grep access_token | wc -l) -ne 1 ]]; then
+    echo "Keycloak token is expired"
+    exit 1
+  else
+    echo "Keycloak token is active."
+  fi
+}
+
+function tagAndPushDocker() {
+  export RH_CHE_AUTOMATION_BUILD_TAG=rh-che-automated-build-$(git rev-parse --short HEAD)
+  export CHE_DOCKER_BASE_IMAGE=eclipse/che-server:nightly-centos
+  echo "che-docker-base-image:${CHE_DOCKER_BASE_IMAGE}"
+  docker pull ${CHE_DOCKER_BASE_IMAGE}
+  docker tag ${CHE_DOCKER_BASE_IMAGE} eclipse/che-server:local
+  rm -rf ../dockerfiles/che-fabric8/eclipse-che
+  mkdir ../dockerfiles/che-fabric8/eclipse-che
+  cp -r ../assembly/assembly-main/target/eclipse-che-fabric8-1.0.0-SNAPSHOT/eclipse-che-fabric8-1.0.0-SNAPSHOT/* ../dockerfiles/che-fabric8/eclipse-che/
+  bash ../dockerfiles/che-fabric8/build.sh || exit 1
+  docker rmi eclipse/che-server:local
+  docker rmi eclipse/che-server:nightly-centos
+  docker tag eclipse/che-server:nightly rhcheautomation/${RH_CHE_AUTOMATION_BUILD_TAG}
+  docker rmi eclipse/che-server:nightly
+  docker login -u ${RH_CHE_AUTOMATION_DOCKERHUB_USERNAME} -p ${RH_CHE_AUTOMATION_DOCKERHUB_PASSWORD}
+  docker push rhcheautomation/che-server:${RH_CHE_AUTOMATION_BUILD_TAG}
+}
+
+# Getting dependencies ready
+sudo yum install dnf
+sudo dnf install --skip-broken --assumeyes \
+            docker \
+            git \
+            make \
+            jq \
+            origin-clients \
+            python3
+systemctl start docker
```

Comment:
Am I mistaken that we will run docker daemon as a part of a CI job? Who will shut it down after script execution ends?

Reply:
I think the nodes are cleanly wiped every time the task is started on them.
Is there some documentation available with details about what is available on the nodes and which services are running?

Reply:
I don't know whether we have such docs